src/stk/io/nrrd.cpp

```diff
@@ -180,29 +181,111 @@ namespace nrrd {
         for (int i = 0; i < 3; ++i) {
             double spacing_i;
             double spacing_dir[NRRD_SPACE_DIM_MAX];
-            int status = nrrdSpacingCalculate(nimg, domain_axis_idx[i], &spacing_i, 
-                spacing_dir);
+            int status = nrrdSpacingCalculate(nimg, domain_axis_idx[i], &spacing_i, spacing_dir);
 
             switch (status) {
-            case nrrdSpacingStatusUnknown:
+            case nrrdSpacingStatusNone: // There is no spacing or direction information
+            case nrrdSpacingStatusUnknown: // The spacing is invalid
                 LOG(Warning) << "nrrd: Could not extract spacing[" << i << "] from " << filename;
                 break;
+
             case nrrdSpacingStatusScalarNoSpace:
+                // The spacing is set, the direction is not defined
+                if (AIR_EXISTS(spacing_i)) {
+                    spacing[i] = (float)spacing_i;
+                }
+                break;
+
             case nrrdSpacingStatusScalarWithSpace:
+                // The spacing is set, the direction is defined but it is missing for this axis
+                LOG(Warning) << "nrrd: Could not extract direction[" << i << "] from " << filename;
+                if (AIR_EXISTS(spacing_i)) {
+                    spacing[i] = (float)spacing_i;
+                }
+                break;
+
             case nrrdSpacingStatusDirection:
+                // Both spacing and direction are present
+                // The call returned the spacing and a unit direction vector
                 if (AIR_EXISTS(spacing_i)) {
                     spacing[i] = (float)spacing_i;
                 }
+                if (AIR_EXISTS(spacing_dir[0]) &&
+                    AIR_EXISTS(spacing_dir[1]) &&
+                    AIR_EXISTS(spacing_dir[2]))
+                {
+                    switch (nimg->space) {
+                    case nrrdSpaceRightAnteriorSuperior:
+                        spacing_dir[0] = -spacing_dir[0];
+                        spacing_dir[1] = -spacing_dir[1];
+                        break;
+                    case nrrdSpaceLeftAnteriorSuperior:
+                        spacing_dir[0] = -spacing_dir[0];
+                        break;
+                    }
+                    direction(0, i) = spacing_dir[0];
+                    direction(1, i) = spacing_dir[1];
+                    direction(2, i) = spacing_dir[2];
+                }
                 break;
             }
         }
 
         Volume vol(size, voxel_type);
         vol.set_origin(origin);
         vol.set_spacing({spacing[0], spacing[1], spacing[2]});
+        vol.set_direction(direction);
 
         memcpy(vol.ptr(), nimg->data, nrrdElementSize(nimg) * nrrdElementNumber(nimg));
 
+        // Measurement frame
+        if(ncomp == 3 && AIR_EXISTS(nimg->measurementFrame[0][0])) {
```

Comment:
Apparently, NRRD vector images can have an additional coordinate transform from the coordinate system of the vector data to world coordinates. Afaik ITK does not have a field like this. It is quite annoying to handle an additional coordinate transform internally for a feature that is likely to be of little use, so I am just converting the coordinates here (i.e. turning the frame to the identity). Not an elegant solution, but I guess it is better than nothing.

I think is is mostly related to tensor imaging, but just in case someone is crazy enough to store displacements with components in a different coordinate system than the image ones, we won't get caught by surprise...